src/coefficients/stdfunction_coefficient.hpp

```diff
@@ -29,29 +30,55 @@ namespace serac {
  */
 class StdFunctionCoefficient : public mfem::Coefficient {
 public:
+
   /**
-   * @brief Constructor that takes in an mfem Vector representing the coordinates and produces a double
+   * @brief Constructor that takes in a time-independent function of space that returns a double
    *
-   * @param[in] func The scalar-returning std::function to define the coefficient
+   * @param[in] func: a function of space and time that returns a double
    */
   StdFunctionCoefficient(std::function<double(mfem::Vector&)> func);
 
   /**
-   * @brief Evalate the coefficient at a quadrature point
+   * @brief Constructor that takes in a time-dependent function of space that returns a double
+   *
+   * @param[in] func: a function of space and time that returns a double
+   */
+  StdFunctionCoefficient(std::function<double(mfem::Vector&, double)> func);
+
+  /**
+   * @brief Evaluate the coefficient at a quadrature point
    *
    * @param[in] Tr The element transformation for the evaluation
    * @param[in] ip The integration point for the evaluation
    * @return The value of the coefficient at the quadrature point
    */
   virtual double Eval(mfem::ElementTransformation& Tr, const mfem::IntegrationPoint& ip);
+  
+  bool is_time_dependent() const { return is_time_dependent_; } 
+
+  friend StdFunctionCoefficient d_dt(const StdFunctionCoefficient &, const double);
+  friend StdFunctionCoefficient d2_dt2(const StdFunctionCoefficient &, const double);
 
 private:
   /**
    * @brief The function to evaluate for the coefficient
    */
-  std::function<double(mfem::Vector&)> func_;
+  std::function<double(mfem::Vector&, double)> func_;
+  bool is_time_dependent_;
 };
 
+inline StdFunctionCoefficient d_dt(const StdFunctionCoefficient & y, const double dt = 1.0e-8) {
+  return StdFunctionCoefficient([dt, y = y.func_](mfem::Vector& x, double t) {
+    return (y(x, t + dt) - y(x, t - dt)) / (2.0 * dt);
+  });
+}
+
+inline StdFunctionCoefficient d2_dt2(const StdFunctionCoefficient & y, const double dt = 1.0e-4) {
+  return StdFunctionCoefficient([dt, y = y.func_](mfem::Vector& x, double t) {
+    return (y(x, t + dt) - 2 * y(x, t) + y(x, t - dt)) / (dt * dt);
+  });
+}
+
```

Comment:
I prefer the free function syntax here, because it more closely resembles the mathematical notation. That being said, these are not actually being used in the odes.hpp implementation. This is another possible way to get at the time derivatives, but it seemed harder to integrate with the boundary condition manager.

Reply:
Looks good to me. Would be nice to have some doxygen comments.